idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -273,65 +232,35 @@ def build(self):
-                        doc='A Pressure Difference'
-                        'for Finite Difference Derivative Calculation',
-                        units=pyunits.Pa)
-        self.efficiency_mech = Param(default=0.97,
-                                     doc="Mechanical Efficiency")
-        self.eff_drive = Param(default=1.0,
-                               doc="Driver efficiency")
+        # Declare variables for the model
+        self.z_s = Var(initialize=0.97373,
+                       doc="Compressibility factors at suction")
+        self.z_d1 = Var(initialize=0.88949,
+                        doc="variable used for calculating"
+                        "compressibility factor at discharge pressure")
+        self.efficiency_mech = Var(initialize=0.97,
+                                   doc="Mechanical Efficiency")
+        self.eff_drive = Var(initialize=1.0,
+                             doc="Driver efficiency")
+        self.r2 = Var(initialize=0.075,
+                      doc="Impeller Tip Radius",
+                      units=pyunits.m)
 
         ############################################################
 
         self.ratioP[:] = 2.0   # make sure these have a number value
         self.deltaP[:] = 0     # to avoid an error later in initialize
 
         ###########################################################
-        # Speed of sound calculations
-        # Finite difference derivative for Kappa_T calculation
-        @self.Constraint(self.flowsheet().config.time, doc="Isothermal "
-                         "Compressibility[m2/N]")
-        def kappaT_con(b, t):
-            Ps = properties_in[t].pressure
-            P0_f = Ps + b.dp
-            rho_v0 = properties_in[t].dens_mol
-            rho_v0f = (P0_f / Ps) * rho_v0
-            return b.kappaT[t] * b.dp == 1 - rho_v0 / rho_v0f
-
-        @self.Constraint(self.flowsheet().config.time,
-                         doc="Speed of Sound at Inlet Condition")
-        def c0_con(b, t):
-            Cp0 = properties_in[t].cp_mol
-            Cv0 = properties_in[t].cv_mol
-            mw = properties_in[t].mw
-            rho_v0 = properties_in[t].dens_mol
-            return b.c0[t]**2 * (Cv0 * mw * b.kappaT[t] * rho_v0) == Cp0
-
-        # --------------------------------------------------------------------
-        # Tip speed, mass flow coefficient, Mach number, and pressure ratio
+        # Calculate Mach number and tip impeller speed
         @self.Constraint(self.flowsheet().config.time, doc="Mach Number")
         def Ma_con(b, t):
-            return b.Ma[t] == b.U2[t] / b.c0[t]
-
-        @self.Constraint(self.flowsheet().config.time, doc='Pressure Ratio')
-        def Pratio_con(b, t):
-            return properties_out[t].pressure == b.ratioP[t] * \
-                  properties_in[t].pressure
+            # assume single vapor phase.
+            speed_of_sound = self.control_volume.properties_in[
+                t].speed_sound_phase['Vap']
```

Comment:
Make sure you document that this model assumes a single vapor phase named "Vap" - IDAES lets a user use whatever name they wish.

Reply:
I wonder if this should be a config option.  Could default to "Vap", but could specify something else.  The phase will always be vapor, but there is a chance the property package doesn't call it "Vap."  That would also serve as partial documentation of it.

Reply:
An alternative would be iterate through all phases and check to see it it is an instance of VaporPhase.

Reply:
@eslickj : I like this idea and it is useful if the property package can handle two-phase inlet stream. Is it correct?
@andrewlee94 : can you point me out how to proceed the iteration? Do I need to declare the VaporPhase and LiquidPhase to iterate it? Please let me know. Thanks.

Reply:
Phases are decaled by the property package:

```
for p in self.config.property_package.phase_list:
    pobj = self.config.property_package.get_phase(p)
    if isinstance(pobj, VaporPhase):
        return p
```

Reply:
Thanks @andrewlee94 . I think this method does not work when I use with the swCO2 property package. Instead, I will save that with another VLE property package. Please check the revised model again. Thanks.